### PR TITLE
Change to onPremise for all CWAgent component binaries and add executable path for Darwin

### PR DESCRIPTION
**Pull request by khanhntd**
# Description of the issue
There are two changes:
- Add executable path for Darwin since it confuses for the customer when using Darwin but [their executable config json for using is Linux as default for other OS.](https://github.com/aws/amazon-cloudwatch-agent/blob/master/translator/cmdutil/translatorutil.go#L74-L76)
- Change config-translator, config-downloader's `onPrem` mode to `onPremise` mode since it would consistent with [`amazon-cloudwatch-agent-ctl` binary](https://github.com/aws/amazon-cloudwatch-agent/blob/master/packaging/dependencies/amazon-cloudwatch-agent-ctl#L64) and aws-otel-collector is using [`onPremise` too](https://github.com/aws-observability/aws-otel-collector/blob/86d542ef6c9afbab2f5fc94d52a1a2e1f7b839c2/tools/ctl/linux/aws-otel-collector-ctl#L178-L189).

# License
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

# Tests
- For the second changes, use `go run config-translator` with mode onPrem and mode onPremise; to be more sure, `make build` and test their binaries.

### CODE REVIEW

File: packaging/darwin/amazon-cloudwatch-agent-ctl
Comment (SaxyPandaBear) [inline packaging/darwin/amazon-cloudwatch-agent-ctl:178]: Where in our code do we validate the mode parameter? What happens if the user passes in an invalid mode? Do we emit any log message anywhere for that? Since this is in the control script, do we want to gracefully recover and continue or fail fast if the mode is invalid?

Comment (khanhntd) [reply packaging/darwin/amazon-cloudwatch-agent-ctl:178]: I would prefer fail fast since it relates to control script if the mode is invalid and besides, I have not seen any recover script after error is found though. Another thing that bothers me besides your questions: if the mode is invalid or somehow the config does not work on that particular mode, can we set mode to `auto` automatically to get the right mode for agent?

File: translator/cmdutil/translatorutil.go
Comment (khanhntd) [reply translator/cmdutil/translatorutil.go]: Agree on this part too. Two cent feelings. Therefore, change to use the default config for linux.

File: packaging/darwin/amazon-cloudwatch-agent-ctl
Comment (SaxyPandaBear) [inline packaging/darwin/amazon-cloudwatch-agent-ctl:170]: As of right now, I think that this should not be removed from the ctl script, although I don think that the default mode should be `auto`, not `ec2`

**Review by SaxyPandaBear — CHANGES_REQUESTED** [review]

File: packaging/darwin/amazon-cloudwatch-agent-ctl
Comment (khanhntd) [reply packaging/darwin/amazon-cloudwatch-agent-ctl:178]: True. The log message are missing the `config.ModeAuto`. Will change the log messages.

Comment (khanhntd) [reply packaging/darwin/amazon-cloudwatch-agent-ctl:178]: No I have not removed the validation in the ctl script. It has a duplicated validation script. The flow would be like this:
1. User inputs their values, if they are not input for flag --mode , the default would be `auto`. If they input besides `auto, ec2, onPremise `, it would [cause an error](https://github.com/aws/amazon-cloudwatch-agent/pull/417/files#). This is the first validation 
2. After the corresponding action such as `fetch-config`, it will go through the `cwa_config` function and mets a second validation, which is the similar to the first one. 
So I don't see any reason to duplicate the validation if it does not use else where besides the ctl.

File: translator/context/context.go
Comment (SaxyPandaBear) [inline translator/context/context.go:117]: This could be reverted

Comment (khanhntd) [reply translator/context/context.go:117]: The comma suits the statements so I guess we don't need to revert it though.

**Comment by SaxyPandaBear** [comment]
LGTM but before I approve, can you double check if there's any public documentation that uses `onPrem` that we would need to update?

**Review by SaxyPandaBear — APPROVED** [review]
I took a look and I think we only have docs on CWA for Linux/Windows that explicitly call out using the ctl script, so looks fine to me

**Comment by khanhntd** [comment]
@SaxyPandaBear I don't see any executables that using -mode onPrem besides the ctl script though. Here are some documents for future reference: 
* https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/install-CloudWatch-Agent-on-premise.html
* https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/install-CloudWatch-Agent-commandline-fleet.html
P/S: EKS, ECS and Cloudformation are using CWAgent's image so we can remove our mind on it.

**Pull request merged**
